frontend/src/components/search/components/video-card/video-card.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC, VideoCard as VideoCardType } from 'common/types/types';
+import { IconName } from 'common/enums/components';
+import { Icon } from 'components/common/common';
+import { useNavigate } from 'hooks/hooks';
+import { getHowLongAgoString } from 'helpers/helpers';
+import { VideoTag } from '../common/common';
+import styles from './styles.module.scss';
+
+type Props = {
+  video: VideoCardType;
+};
+
+const VideoCard: FC<Props> = ({ video: { id, name, duration, videoViews, createdAt, preview, channel } }) => {
+  const navigate = useNavigate();
+
+  const redirectToVideoPage = (): void => navigate(`/video/${id}`, { replace: true });
+  const redirectToChannelPage = (): void => navigate(`/channel/${channel.id}`, { replace: true });
+
+  const isNew = (): boolean => {
+    const date = new Date(createdAt);
+    return Date.now() - date.getTime() <= 4 * 60 * 60 * 1000;
+  };
+
+  const getDurationString = (durationInSeconds: number): string => {
+    const hr = Math.floor(durationInSeconds / 3600);
+    const min = Math.floor((durationInSeconds % 3600) / 60);
+    const sec = Math.floor(durationInSeconds % 60);
+
+    return `${hr > 0 ? hr + ':' : ''}${hr && min < 10 ? '0' : ''}${min}:${sec < 10 ? '0' : ''}${sec}`;
```

Comment:
how about to craete useful helper

```javascript
export const joinExistingValues = (values: any[], separator: string) => {
  return values.filter(Boolean).join(separator);
};
```

it would be much easier to read your code then